src/tron.jl

```diff
@@ -94,10 +94,14 @@ mutable struct TronSolver{T, V <: AbstractVector{T}, Op <: AbstractLinearOperato
   gpx::V
   Hs::V
   H::Op
-  tr::TrustRegion{T, V}
+  tr::TRONTrustRegion{T, V}
 end
 
-function TronSolver(nlp::AbstractNLPModel{T, V};) where {T, V <: AbstractVector{T}}
+function TronSolver(
+  nlp::AbstractNLPModel{T, V};
+  max_radius::T = min(one(T) / sqrt(2 * eps(T)), T(100)),
```

Comment:
We are disassociating the `max_radius` from the `initial_radius`, but maybe it is best if we accept an additional argument, for `initial_radius`?
Or maybe we should accept `trust_region_kwargs`, like in Percival? We can add `TR <: AbstractTrustRegion{T, V}`.

Reply:
I added a mechanism so that all the kwargs taken by `TRONTrustRegion` are valid now.

Reply:
It is good until generalization